cmd/osbuild-jobsite-builder/main.go

```diff
@@ -206,6 +209,47 @@ func (builder *Builder) HandlePopulate(w http.ResponseWriter, r *http.Request) e
 	if r.Method != "POST" {
 		return fmt.Errorf("Builder.HandlePopulate: unexpected request method")
 	}
+	storePath := path.Join(argBuildPath, "store")
+	err := os.Mkdir(storePath, 0755)
+	if err != nil {
+		return fmt.Errorf("Builder.HandlePopulate: failed to make store directory: %v", err)
+	}
+
+	tarReader := tar.NewReader(r.Body)
+	for header, err := tarReader.Next(); err != io.EOF; header, err = tarReader.Next() {
+		if err != nil {
+			return fmt.Errorf("Builder.HandlerPopulate: failed to unpack sources: %v", err)
+		}
+
+		if !strings.HasPrefix(path.Join(storePath, header.Name), filepath.Clean(storePath)) {
+			return fmt.Errorf("Builder.HandlerPopulate: name not clean: %v doesn't have %v prefix", header.Name, filepath.Clean(storePath))
+		}
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.Mkdir(path.Join(storePath, header.Name), header.FileInfo().Mode()); err != nil {
+				return fmt.Errorf("Builder.HandlerPopulate: unable to make dir in sources: %v", err)
+			}
+		case tar.TypeReg:
+			file, err := os.Create(path.Join(storePath, header.Name))
+			if err != nil {
+				return fmt.Errorf("Builder.HandlerPopulate: unable to open file in sources: %v", err)
+			}
+			defer file.Close()
+			for {
+				_, err := io.CopyN(file, tarReader, 4096)
```

Comment:
Why not just `io.Copy()` here? Is the concern  the buffersize of io.Copy() ? If so, maybe io.CopyBuffer but it should be okay, iirc io.Copy() uses a 32kb internal buffer which is not amazing but should be okay? or am I missing something?

Reply:
gosec complains :(

Reply:
https://pkg.go.dev/github.com/securego/gosec/v2#section-readme g110 specifically

Reply:
Thanks! I see, idea below. But I also think that gosec is silly here, we trust the inputs and also we cannot know how big or small a file in the tar should be so it's hard to guard against this.

maybe just: 
```go
diff --git a/cmd/osbuild-jobsite-builder/main.go b/cmd/osbuild-jobsite-builder/main.go
index 9037a6601..7e880b6ae 100644
--- a/cmd/osbuild-jobsite-builder/main.go
+++ b/cmd/osbuild-jobsite-builder/main.go
@@ -239,16 +239,12 @@ func (builder *Builder) HandlePopulate(w http.ResponseWriter, r *http.Request) e
                        }
                        defer file.Close()
                        // copy in defined chunks to avoid gosec#G110
-                       for {
-                               _, err := io.CopyN(file, tarReader, 4096)
-                               if err != nil {
-                                       if err == io.EOF {
-                                               break
-                                       }
-                                       return fmt.Errorf("Builder.HandlerPopulate: unable to write file in sources: %v", err)
-                               }
+                       if _, err := io.CopyN(file, tarReader, header.Size); err != nil {
+                               return fmt.Errorf("Builder.HandlerPopulate: unable to write file in sources: %v", err)
+                       }
+                       if err := file.Close(); err != nil {
+                               return fmt.Errorf("Builder.HandlerPopulate: unable to close file in sources: %v", err)
                        }
-                       file.Close()
                default:
                        return fmt.Errorf("Builder.HandlerPopulate: unexpected tar header type: %v", header.Typeflag)
                }

````
to avoid the loop? internally CopyN is just Copy with a limited reader so that should be fine (not tested but happy to write a test).

Reply:
we can also just turn it off, maybe that's best. Will do that